cmd/skaffold/app/cmd/cmd_test.go

```diff
@@ -50,3 +51,44 @@ func TestPreReleaseVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateCheck(t *testing.T) {
+	tests := []struct {
+		description string
+		versionStr  string
+		expected    string
+		versionMsg  string
+		updateCheck bool
+	}{
+		{
+			description: "pre release version, update check is disabled",
+			versionStr:  "v1.20.0-42-g92900f245-dirty",
+		},
+		{
+			description: "release version but update check disabled",
+			versionStr:  "v1.20.0",
+		},
+		{
+			description: "release version and update check enabled",
+			versionStr:  "v1.20.0",
+			updateCheck: true,
+			versionMsg:  "newer version is available",
+			expected:    "newer version is available",
+		},
+		{
+			description: "update check enabled but version already up to date",
+			versionStr:  "v1.20.0",
+			updateCheck: true,
+		},
```

Comment:
Our bleeding edge binaries have a version with the abbreviated SHA.  Could you add a test for that too?
```
$ ./skaffold version
36395cc
```